modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -145,9 +145,91 @@ class SoftMaxSubgraph : public Subgraph
+        int inpShape1 = addNodeToMatch("Shape", input);
+        int inpGather1 = addNodeToMatch("Gather", inpShape1, indices1);
+        int inpCastG1 = addNodeToMatch("Cast", inpGather1);
+        int B1 = addNodeToMatch("Constant");
+        int inpMul1 = addNodeToMatch("Mul", inpCastG1, B1);
+        int inpCastM1 = addNodeToMatch("Cast", inpMul1);
+        int inpFloor1 = addNodeToMatch("Floor", inpCastM1);
+
+        int indices2 = addNodeToMatch("Constant");
+        int inpShape2 = addNodeToMatch("Shape", input);
+        int inpGather2 = addNodeToMatch("Gather", inpShape2, indices2);
+        int inpCastG2 = addNodeToMatch("Cast", inpGather2);
+        int B2 = addNodeToMatch("Constant");
+        int inpMul2 = addNodeToMatch("Mul", inpCastG2, B2);
+        int inpCastM2 = addNodeToMatch("Cast", inpMul2);
+        int inpFloor2 = addNodeToMatch("Floor", inpCastM2);
+
+        int inpUnsqueeze1 = addNodeToMatch("Unsqueeze", inpFloor1);
+        int inpUnsqueeze2 = addNodeToMatch("Unsqueeze", inpFloor2);
+        int inpConcat1 = addNodeToMatch("Concat", inpUnsqueeze1, inpUnsqueeze2);
+
+        int inpIndex = addNodeToMatch("Constant");
+
+        int inpCast3 = addNodeToMatch("Cast", inpConcat1);
+
+        int inpShape3 = addNodeToMatch("Shape", input);
+        int inpSlice = addNodeToMatch("Slice", inpShape3);
+        int inpCast4 = addNodeToMatch("Cast", inpSlice);
+
+        int inpDiv = addNodeToMatch("Div", inpCast3, inpCast4);
+        int inpConcat2 = addNodeToMatch("Concat", inpIndex, inpDiv);
+
+        addNodeToMatch("Upsample", input, inpConcat2);
+        setFusedNode("FusedUpsample", input, indices1, B1, indices2, B2, inpIndex);
+    }
+
+    virtual bool match(const Ptr<ImportGraphWrapper>& net, int nodeId,
+                       std::vector<int>& matchedNodesIds,
+                       std::vector<int>& targetNodesIds) CV_OVERRIDE
+    {
+        if (Subgraph::match(net, nodeId, matchedNodesIds, targetNodesIds))
+        {
+            Ptr<ImportNodeWrapper> upsample = net->getNode(matchedNodesIds[21]);
+            opencv_onnx::NodeProto* node = upsample.dynamicCast<ONNXNodeWrapper>()->node;
+
+            CV_Assert(node->op_type() == "Upsample");
+
+            for (int i = 0; i < node->attribute_size(); i++)
+            {
+                opencv_onnx::AttributeProto attr = node->attribute(i);
+                if (attr.name() != "mode")
+                    continue;
+                mode = attr.s();
+                return true;
+            }
+        }
+        return false;
+    }
+
+    virtual void finalize(const Ptr<ImportGraphWrapper>&,
+                          const Ptr<ImportNodeWrapper>& fusedNode,
+                          std::vector<Ptr<ImportNodeWrapper> >&) CV_OVERRIDE
+    {
+        opencv_onnx::NodeProto* node = fusedNode.dynamicCast<ONNXNodeWrapper>()->node;
+        opencv_onnx::AttributeProto* attr = node->add_attribute();
+        attr->set_name("mode");
+        attr->set_s(mode);
```

Comment:
It seems to me that fused node is a last node in subgraph (Upsample) so it should already has this attribute

Reply:
yeah, you were right. Thanks.